src/script/api/script_group.cpp

```diff
@@ -150,21 +150,27 @@
 	return ScriptObject::DoCommand(0, group_id << 16, (::INVALID_ENGINE << 16) | engine_id, CMD_SET_AUTOREPLACE);
 }
 
-/* static */ Money ScriptGroup::GetProfitThisYear(GroupID group_id)
+template <typename T, typename Func>
+T SumVehiclesInGroup(GroupID group_id, Func func)
 {
-	if (!IsValidGroup(group_id)) return -1;
-
-	Money profit = 0;
+	T result{};
 
 	const Vehicle *v;
 	FOR_ALL_VEHICLES(v) {
 		if (v->group_id != group_id) continue;
 		if (!v->IsPrimaryVehicle()) continue;
 
-		profit += v->GetDisplayProfitThisYear();
+		result += func(v);
 	}
 
-	return profit;
+	return result;
+}
+
+/* static */ Money ScriptGroup::GetProfitThisYear(GroupID group_id)
+{
+	if (!IsValidGroup(group_id)) return -1;
+
+	return SumVehiclesInGroup<Money>(group_id, [](const Vehicle * v) { return v->GetDisplayProfitThisYear(); });
```

Comment:
extra space

```suggestion
	return SumVehiclesInGroup<Money>(group_id, [](const Vehicle *v) { return v->GetDisplayProfitThisYear(); });
```

same in a few other places